Lib/ssl.py

```diff
@@ -994,9 +994,38 @@ def _create(cls, sock, server_side=False, do_handshake_on_connect=True,
             if e.errno != errno.ENOTCONN:
                 raise
             connected = False
+            blocking = self.getblocking()
+            self.setblocking(False)
+            try:
+                # We are not connected so this is not supposed to block, but
+                # testing revealed otherwise on macOS and Windows so we do
+                # the non-blocking dance regardless. Our raise when any data
+                # is found means consuming the data is harmless.
+                notconn_pre_handshake_data = self.recv(1)
+            except OSError as e:
```

Comment:
This shadows the existing variable name `e`, which is used below in `SSLError(e.errno, reason)`. The control flow is such that this does not matter as it only shadows it if an exception occurs, in which case `notconn_pre_handshake_data` is set to `b''` and the `if notconn_pre_handshake_data:` block will not be executed, but it did make the code harder to understand for me, and could cause a bug if the control flow ever changes.

I also wonder if the fact that the original variable is apparently deleted after being shadowed in the nested `except` block is intended behaviour or a Python bug:

```py
>>> try:
...     raise Exception("oops")
... except Exception as e:
...     print("A", e)
...     try:
...         raise Exception("uh-oh")
...     except Exception as e:
...         print("B", e)
...     print("C", e)
...
A oops
B uh-oh
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
Exception: oops

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 9, in <module>
NameError: name 'e' is not defined
```

Reply:
thanks for the extra eyeballs reviewing our change regardless!

Reply:
No problem! I do still feel like the variable shadowing makes the code harder to follow, which was what I wanted to point out when I got a bit sidetracked by the *to me* unexpected intended behaviour of the except clause :)